src/recordlinker/schemas/pii.py

```diff
@@ -205,22 +208,38 @@ class Address(StrippedBaseModel):
     latitude: typing.Optional[float] = None
     longitude: typing.Optional[float] = None
 
-    @functools.cached_property
-    def normalize_state(self) -> str | None:
+    @pydantic.field_validator("line", mode="before")
+    def parse_line(cls, value: list[str]) -> list[str]:
         """
-        Normalize the state field into 2-letter USPS code.
+        Parse the line field into a list of strings with normalized street suffixes.
         """
+        normalized = []
+        for line in value:
+            parts = line.strip().split(" ")
+            # remove all non-alphanumeric characters and convert to uppercase
+            suffix = "".join(c for c in parts[-1] if c.isalnum()).upper()
+            if common := cls.ST_SUFFIXES.get(suffix):
+                # replace the suffix with the common suffix
+                parts[-1] = common
+            normalized.append(" ".join(parts))
+        return normalized
 
-        if self.state:
-            state = self.state.strip().title()
+    @pydantic.field_validator("state", mode="before")
+    def parse_state(cls, value: str) -> str | None:
+        """
+        Normalize the state field into 2-letter USPS code.
+        """
+        if value:
+            state = value.strip().title()
+            # reduce inner whitespace to a single whitespace char
+            state = " ".join(w for w in state.split(" ") if w)
 
             if len(state) == 2 and state.upper() in _STATE_CODE_TO_NAME:
-                return self.state.upper()
+                return state.upper()
 
             if state in _STATE_NAME_TO_CODE:
                 return _STATE_NAME_TO_CODE[state]
-
-        return None
+        return value
```

Comment:
@m-goggins @bamader I'm not married to this change.  I've been going back and forth on ignoring the state value if it can't be normalized to a two char USPS code or keeping it as is.  I think with the latter, at least we're preserving something for fuzzy evaluation if there was a typo in the spelling.

Reply:
I agree as well, that's still meaningful data to keep around